### DISCUSSION TITLE
DeepMD-ZBL Shifting Energy by energy of atom in vacuum

### DISCUSSION THREAD

**Discussion by JRVHouston628** [deepmodeling/deepmd-kit#2662] [Q&A]
Hello,

I'm trying to use the DeepMD-ZBL methodology referenced here: https://arxiv.org/pdf/1904.00360.pdf for Si irradiation simulations using the use_srtab command. I am using QuantumEspresso to train the model. Do I need to ensure the energy documented in the tabulated file converges to the energy of a Si in a vacuum as calculated by Quantum Espresso at large distances? That is, does the ZBL potential need to be shifted down?

**Comment by wanghan-iapcm** [comment]
No. The tab potential only switches on when the neighbors are very close to an atom. It has no effect on the asymptotic behavior at large distance.

**Reply by JRVHouston628** [reply]
Yes I realize that. I don't think I worded my question correctly.
If I construct a table consisting of the interatomic distance in column 1 and the energy from the ZBL potential in column 2, using the classical definition of the ZBL potential, it should eventually decay to an energy of 0. That is we consider an atom in a vacuum to have zero energy.

However, in this case we are trying to stitch ZBL to DeepMD training on Quantum Espresso calculations. The QE calculations that the energy of an atom in a vacuum is not 0 (as is expected).

My question is: do I have to make sure the ZBL potential in the table decays to the energy indicated by QE calculations in order to be consistent?

**Reply by wanghan-iapcm** [reply]
Thanks a lot for the detailed explanation and that is clear to me. You made a very good point. I think one should shift the ZBL to the same energy as the DFT code. 
What the deepmd-kit does is that it smoothly interpolates the zbl and DP in the interval [`sw_min`, `sw_max`]. Suppose that the vacuum energy deviate too much from 0 and the ZBL is not shifted, then although the resultant potential is still smooth, it may be ill-behaved in the interpolation interval as it tries to interpolate a large energy jump.

**Reply by njzjz** [reply]
To my understanding, the current tab potential only has two-atom interactions and does not contribute atomic energy. This might be something that needs to be improved.

**Reply by njzjz** [reply]
@wanghan-iapcm  #2670 should work?

**Reply by njzjz** [reply]
I feel a bit confused here. Since the atomic energy in a vacuum does not contribute to forces, can the energy match the forces in the old model? It seems to me that the total energy might not be conserved when performing an NVE simulation. (But since the forces are correct, the trajectory should be correct)

**Reply by wanghan-iapcm** [reply]
The deepmd-kit uses interpolation between potential energies. the corresponding force scheme is the interpolation of forces plus a term proportional to the energy difference. see the attached figure which was developed for pair-wise potential. Therefore, the energy jump would introduce an ill-behaved force. 
![2023-07-12-080632_1643x419_scrot](https://github.com/deepmodeling/deepmd-kit/assets/92130845/6960c6ac-d573-49ea-bb33-08ad057def66)

**Reply by njzjz** [reply]
I got your meaning.

**Reply by wanghan-iapcm** [reply]
An side-effect introduced by shifting the ZBL should be noticed. 

The tabulated pair-wise potential always assumes that the vacuum energy is 0, thus by shifting a discontinuity is introduced at the cut-off distance. 

Therefore, when the ZBL is switched on, the energy is not conserved as one particle passes through the cut-off. The force is not affected by the discontinuity.

**Reply by JRVHouston628** [reply]
I will mention that I have attempted to use this methodology to develop of Si potential in order to study Si self sputtering.
I took two approaches when training the model:
1. I shifted the ZBL energy so that it would converge to the zero-point energy calculated from quantum espresso at infinite distances.
2. I did not shift the ZBL energy.

In the first case, the simulation cell was completely unstable when it was impacted by an incoming Si atom. The second case remained stable.

**Reply by wanghan-iapcm** [reply]
I would prefer shifting the DFT energy zero point, but not shifting the ZBL potential. @JRVHouston628 have you tried this option?

**Reply by JRVHouston628** [reply]
I haven't tried this yet, but I think it's worth a try. Let me spend some time on this and I'll get back to everyone.

**Reply by JRVHouston628** [reply]
[DeepMD_ZBL.docx](https://github.com/deepmodeling/deepmd-kit/files/12102113/DeepMD_ZBL.docx)

Hi all,
I still have not tried shifting the DFT zero point energy as of yes, but I wanted to attach a document summarizing my findings that I mentioned.  I show the potential energy of an Si-Si dimer for two cases: 

The first case is using a DeepMD+ZBL potential where the tabulated potential was shifted down to match the calculated zero-point energy. The potential energy curve appears smooth and well behaved, however when I hit the surface with a Si atom (with a kinetic energy of 100 eV), the entire simulation cell explodes.

The second case is using a DeepMD+ZBL potential where the tabulated potential was **NOT** shifted down to match the calculated zero-point energy. The potential energy curve shows some odd behavior (as expected) in the region where smoothing occurs (between 1 and 1.6 angstroms). However despite this apparent oddity in the dimer potential energy, the simulation cell remains stable when hit with a Si atom with kinetic energy of 100 eV.

I would have thought the first case would be more well behaved given how smooth the dimer potential energy looks, however the opposite is true.  Can someone explain why this is happening? I was under the impression that only the forces are important, but it appears whether or not you shift your ZBL potential significantly affects results.

**Reply by wanghan-iapcm** [reply]
We have supported shifting the tabulated potential, please check https://github.com/deepmodeling/deepmd-kit/pull/2670

**Reply by JRVHouston628** [reply]
Thanks I'll give this a try soon! I appreciate the help!
